src/cascade/model/model_reader.py

```diff
@@ -0,0 +1,313 @@
+from functools import partial
+from math import isnan
+
+import numpy as np
+import pandas as pd
+
+from cascade.core import getLoggers
+from cascade.dismod.constants import IntegrandEnum, RateEnum, PriorKindEnum
+from cascade.model.age_time_grid import AgeTimeGrid
+from cascade.model.dismod_groups import DismodGroups
+from cascade.model.var import Var
+
+CODELOG, MATHLOG = getLoggers(__name__)
+
+
+def write_vars(dismod_file, new_vars, var_ids, which):
+    """
+
+    Args:
+        dismod_file:
+        new_vars (DismodGroups): The new vars to write.
+        var_ids (DismodGroups): The output of ``read_var_table_as_id``.
+        which (str): Could be "start_var", "truth_var", "scale_var", "fit_var".
+    """
+    var_name = f"{which}_var"
+    id_column = f"{var_name}_id"
+    total = list()
+    for group_name, group in var_ids.items():
+        for key, value in group.items():
+            total.append(_construct_vars_one_field(var_name, value, new_vars[group_name][key]))
+    new_table = pd.concat(total).sort_values(by=[id_column])
+    setattr(dismod_file, var_name, new_table)
+
+
+def _construct_vars_one_field(name, var_id, new_var):
+    id_column = f"{name}_id"
+    var_column = f"{name}_value"
+    with_id = new_var.grid.merge(var_id.grid[["age", "time", "var_id"]], on=["age", "time"], how="left")
+    if with_id.var_id.isna().any():
+        raise RuntimeError(f"Internal error: Could not align {id_column} with var_id.")
+    return pd.DataFrame({
+        id_column: with_id.var_id.values,
+        var_column: with_id["mean"].values,
+    })
+
+
+def read_vars(dismod_file, var_ids, which):
+    """
+    Reads a table full of vars (start, truth, scale).
+
+    Args:
+        dismod_file:
+        var_ids (DismodGroups): The output of ``read_var_table_as_id``.
+        which (str): Could be "start_var", "truth_var", "scale_var", "fit_var".
+
+    Returns:
+        DismodGroups: With means for everything.
+    """
+    var_name = f"{which}_var"
+    table = getattr(dismod_file, var_name)
+    return _assign_from_var_ids(table, var_ids, partial(_read_vars_one_field, name=var_name))
+
+
+def _assign_from_var_ids(table, var_ids, assignment):
```

Comment:
I find the name `assignment` to be very confusing. I actually spent quite a while looking for where it was defined when I saw it down below. Then I realized it was an argument to the function. I think I would have been less confused with the generic but conventional `func`. Also is there a reason not to call this `_map_over_var_ids` or something? It actually doesn't do any assignment, it returns a new value.

Reply:
Yup. Yup. Yup. Highly misleading.